### PR TITLE
Filter out credentials in stderr message 

### LINKED ISSUE #4663: Filter out credentials for git sources in stderr output
- Related to #4425

While Bundler will screen out sensitive credentials for stdout output for git sources, it will not do the same for stderr output at the moment.

patch for failing test.
```
diff --git a/spec/install/gemfile/git_spec.rb b/spec/install/gemfile/git_spec.rb
index 2cb62b4..fd0da26 100644
--- a/spec/install/gemfile/git_spec.rb
+++ b/spec/install/gemfile/git_spec.rb
@@ -1116,6 +1116,7 @@ describe "bundle install with git sources" do
 
         bundle :install, :expect_err => true
         expect(out).to_not include("password1")
+        expect(err).to_not include("password1")
         expect(out).to include("Fetching https://user1@github.com/company/private-repo")
       end
     end
@@ -1132,6 +1133,7 @@ describe "bundle install with git sources" do
 
         bundle :install, :expect_err => true
         expect(out).to_not include("oauth_token")
+        expect(err).to_not include("oauth_token")
         expect(out).to include("Fetching https://x-oauth-basic@github.com/company/private-repo")
       end
     end
```

### PR DESCRIPTION
**Pull request by allenzhao**
This will close #4663 
- [x] Fix broken tests

### CODE REVIEW

File: lib/bundler/source/git/git_proxy.rb [hunk 1]
```diff
@@ -147,11 +148,14 @@ def git_retry(command)
         def git(command, check_errors = true, error_msg = nil)
           command_with_no_credentials = URICredentialsFilter.credential_filtered_string(command, uri)
           raise GitNotAllowedError.new(command_with_no_credentials) unless allow?
+          std_out, std_err, status = SharedHelpers.with_clean_git_env do
+            Open3.capture3("git #{command}")
```
Comment (segiddins) [inline lib/bundler/source/git/git_proxy.rb]: this might not work on windows with ruby 1.8.7

Comment (allenzhao) [inline lib/bundler/source/git/git_proxy.rb]: I discussed with @RochesterinNYC earlier and we didn't come up with a better idea of how to handle `stderr`

Comment (allenzhao) [inline lib/bundler/source/git/git_proxy.rb]: I will try to test it manually

**Comment by segiddins** [comment]
Is this still a WIP?

**Comment by allenzhao** [comment]
@segiddins Nope, just finished implementing ;)

File: lib/bundler/source/git/git_proxy.rb [hunk 2]
```diff
@@ -220,6 +223,23 @@ def allowed_in_path
           return in_path { yield } if allow?
           raise GitError, "The git source #{uri} is not yet checked out. Please run `bundle install` before trying to start your application"
         end
+
+        def capture_and_filter_stderr(uri)
```
Comment (segiddins) [inline lib/bundler/source/git/git_proxy.rb]: can you please add a TODO to replace this with `Open3` in bundler 2?

**Comment by segiddins** [comment]
@homu delegate=RochesterinNYC

**Comment by indirect** [comment]
@homu r+

**Pull request merged**